### PR TITLE
[LM-10562] Added cohort and channel support

### PR DESCRIPTION
**Pull request by arnavr15**
## Description

**Features**
- Added cohort and channel support

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [ ] Label added
- [x] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [x] Description about the Feature/ Bugs added

### CODE REVIEW

File: example/package.json [hunk 1]
```diff
@@ -34,6 +34,7 @@
     "react-native-blob-util": "^0.19.2",
```
Comment (jaixchaudhary) [inline example/package.json:34]: remove caret from all the dependencies

File: likeminds-chat-reactnative-integration/ChatSX/components/HomeFeedItem/index.tsx [hunk 2]
```diff
@@ -423,28 +430,29 @@ const HomeFeedItem: React.FC<Props> = ({
...
-            ]}>
-            {deletedBy !== 'null' &&
+            ]}
+          >
+            {deletedBy !== "null" &&
             deletedBy !== null &&
             deletedBy !== undefined ? (
               <Text style={styles.deletedMessage}>
-                {'This message has been deleted'}
+                {"This message has been deleted"}
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/components/HomeFeedItem/index.tsx:441]: move it to the strings.ts file and import it from there

File: likeminds-chat-reactnative-integration/ChatSX/components/InputBox/index.tsx [hunk 3]
```diff
@@ -2335,7 +2335,7 @@ const InputBox = ({
                   plainTextStyle={[
                     inputBoxStyles?.plainTextStyle,
                     {
-                      color: !isUploadScreen
+                      color: isUploadScreen
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/components/InputBox/index.tsx:2338]: why this change?

File: likeminds-chat-reactnative-integration/ChatSX/store/actions/explorefeed.ts [hunk 4]
```diff
@@ -11,15 +11,13 @@ import {
 import { Dispatch } from "@reduxjs/toolkit";
 import { Client } from "../../client";
 
-const myClient = Client.myClient;
-
 export const getExploreFeedData =
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/store/actions/explorefeed.ts:14]: revert this change and make a variable myClient and reuse it

File: likeminds-chat-reactnative-integration/ChatSX/store/actions/homefeed.ts [hunk 5]
```diff
@@ -35,14 +35,12 @@ import {
 import { Dispatch } from "@reduxjs/toolkit";
 import { Client } from "../../client";
 
-const myClient = Client.myClient;
-
 export const initAPI = (payload: any) => async (dispatch: Dispatch) => {
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/store/actions/homefeed.ts:38]: revert this change and make a variable myClient and reuse it

File: likeminds-chat-reactnative-integration/ChatSX/store/actions/homefeed.ts [hunk 6]
```diff
@@ -113,7 +111,7 @@ export const updateInvites =
 //     return await dispatch({
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/store/actions/homefeed.ts:111]: remove unnecessary comments

File: likeminds-chat-reactnative-integration/ChatSX/utils/chatroomUtils.ts [hunk 7]
```diff
@@ -3,8 +3,6 @@ import { UserInfo } from "../db/models";
 import { GetConversationsType } from "../enums";
 import { Client } from "../client";
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/utils/chatroomUtils.ts:5]: revert this change and make a variable myClient and reuse it

File: likeminds-chat-reactnative-integration/ChatSX/utils/syncChatroomApi.ts [hunk 8]
```diff
@@ -1,8 +1,6 @@
 import { SyncChatroomRequest } from "@likeminds.community/chat-rn";
 import { Client } from "../client";
 
-const myClient = Client.myClient;
-
 // Sync Chatrrom API
```
Comment (jaixchaudhary) [inline likeminds-chat-reactnative-integration/ChatSX/utils/syncChatroomApi.ts:4]: revert this change and make a variable myClient and reuse it

File: likeminds-chat-reactnative-integration/ChatSX/store/actions/chatroom.ts [hunk 9]
```diff
@@ -23,15 +23,13 @@ import {
 } from "../types/types";
 import { Client } from "../../client";
 
-const myClient = Client.myClient;
-
 export const getConversations =
```
[the remark this answers is not in the document]
Comment (arnavr15) [reply likeminds-chat-reactnative-integration/ChatSX/store/actions/chatroom.ts:26]: chatroom is a file containing methods which are exported from there, so if we declare `const myClient = Client.myClient;` globally, flow will never go there since its not a component and hence `myClient` inside methods will be undefined

File: likeminds-chat-reactnative-integration/ChatSX/components/InputBox/index.tsx (hunk 3 above)
Comment (arnavr15) [reply likeminds-chat-reactnative-integration/ChatSX/components/InputBox/index.tsx:2338]: because this is correct, earlier one was reverse

File: likeminds-chat-reactnative-integration/ChatSX/components/ParseDeepLink/index.ts [hunk 10]
```diff
@@ -40,7 +39,7 @@ export async function parseDeepLink(
           uuid: request?.uuid,
           isGuest: false,
         };
-        const initiateUserResponse = await myClient?.initiateUser(
+        const initiateUserResponse = await Client.myClient?.initiateUser(
```
[the remark this answers is not in the document]
Comment (arnavr15) [reply likeminds-chat-reactnative-integration/ChatSX/components/ParseDeepLink/index.ts:42]: chatroom is a file containing methods which are exported from there, so if we declare const myClient = Client.myClient; globally, flow will never go there since its not a component and hence myClient inside methods will be undefined

**Review by jaixchaudhary — APPROVED** [review]

**Review by jaixchaudhary — APPROVED** [review]
